crates/exec-wasmtime/src/lib.rs

```diff
@@ -105,14 +105,14 @@ mod test {
     )"#;
 
     pub fn run(wasm: &[u8]) -> anyhow::Result<Vec<Val>> {
-        let mut file = NamedTempFile::new().context("failed to create module file")?;
+        let mut file = tempfile().context("failed to create module file")?;
         file.write(wasm).context("failed to write module to file")?;
         file.rewind().context("failed to rewind file")?;
         Runtime::execute(Package::Local {
             #[cfg(unix)]
-            wasm: (&file).as_raw_fd(),
+            wasm: file.as_raw_fd(),
             #[cfg(windows)]
-            wasm: file.into_file(),
```

Comment:
well, I guess, you could have use `NamedTempFile`, if you kept the result of `file.into_file()` in scope as long as the file is needed.
```rust
{ // scope start
   let wasm_file = file.into_file();
   Runtime::execute(Package::Local /* ... */)
} // scope end, wasm_file gets deleted by `drop`
```

Reply:
Yes, but we don't need `NamedTempFile` and `tempfile()` is the preferred option.

> When choosing between the temporary file variants, prefer `tempfile` unless you either need to know the file’s path or to be able to persist it.

https://docs.rs/tempfile/latest/tempfile/index.html#design